##Packages

In [4]:
import pandas as pd
import numpy as np
from sklearn import preprocessing     # For normalizing data

from scipy import stats   # for t-test
from statistics import variance

from scipy.sparse import coo_matrix
from sklearn.utils import shuffle
from sklearn.model_selection import train_test_split
from sklearn.metrics import f1_score

from google.colab import drive        # For mounting google drive for features.csv

##Mount from Google Drive

In [3]:
drive.mount('/content/gdrive')
fish = pd.read_csv("gdrive/My Drive/CSCE624/project/fish1.csv")
general_use = pd.read_csv("gdrive/My Drive/CSCE624/project/general_use.csv")
osu = pd.read_csv("gdrive/My Drive/CSCE624/project/osu1.csv")

Mounted at /content/gdrive


## Data

### Student's T-Test

#### Fish

##### Mouse X

###### vs Right Eye

In [5]:
# Mouse X vs Right Eye
fish_mouse_x = fish[[" mouse_x"]]
fish_right_eye_x = fish[[" right_eye_x"]]

In [6]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
fish_mouse_x_values = fish_mouse_x.values
fish_mouse_x_scaled = min_max_scaler.fit_transform(fish_mouse_x_values)

# right eye x
fish_right_eye_x_values = fish_right_eye_x.values
fish_right_eye_x_scaled = min_max_scaler.fit_transform(fish_right_eye_x_values)

In [7]:
# T-Test
stats.ttest_ind(fish_right_eye_x_scaled, fish_mouse_x_scaled)

Ttest_indResult(statistic=array([30.75149949]), pvalue=array([2.3692734e-170]))

This t-test produces a p-value of **2.3692734e-170**, showing strong correlation between Mouse X and Right Eye X.

In [68]:
# Covariance
fmx_scaled = np.concatenate(fish_mouse_x_scaled, axis=0 )
frex_scaled = np.concatenate(fish_right_eye_x_scaled, axis=0 )
np.corrcoef(frex_scaled, fmx_scaled)

array([[ 1.        , -0.53316202],
       [-0.53316202,  1.        ]])

###### vs Left Eye

In [8]:
# Mouse X vs Left Eye
fish_mouse_x = fish[[" mouse_x"]]
fish_left_eye_x = fish[[" left_eye_x"]]

In [9]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
fish_mouse_x_values = fish_mouse_x.values
fish_mouse_x_scaled = min_max_scaler.fit_transform(fish_mouse_x_values)

# left eye x
fish_left_eye_x_values = fish_left_eye_x.values
fish_left_eye_x_scaled = min_max_scaler.fit_transform(fish_left_eye_x_values)

In [10]:
stats.ttest_ind(fish_left_eye_x_scaled, fish_mouse_x_scaled)

Ttest_indResult(statistic=array([31.22738378]), pvalue=array([1.10808965e-174]))

This t-test produces a p-value of **1.10808965e-174**, showing strong correlation between Mouse X and Left Eye X.

In [67]:
# Covariance
fmx_scaled = np.concatenate(fish_mouse_x_scaled, axis=0 )
flex_scaled = np.concatenate(fish_left_eye_x_scaled, axis=0 )
np.corrcoef(flex_scaled, fmx_scaled)

array([[ 1.        , -0.52731846],
       [-0.52731846,  1.        ]])

In [ ]:
# Sum of Squared Differences
ssd = np.sum((fish_right_eye_x_scaled - fish_left_eye_x_scaled)**2)
print(ssd)

2.3474122546103686


In [ ]:
# Left vs Right
lr_diff = fish_right_eye_x_scaled - fish_left_eye_x_scaled
lr_diff = np.concatenate(lr_diff, axis=0 )
variance(lr_diff)

0.002284914301571051

In [14]:
# Covariance
fish_right_x = np.concatenate(fish_right_eye_x_scaled, axis=0 )
fish_left_x = np.concatenate(fish_left_eye_x_scaled, axis=0 )
np.cov(fish_right_x, fish_left_x)

array([[0.00957358, 0.00952898],
       [0.00952898, 0.01176929]])

In [16]:
# Correlation
np.corrcoef(fish_right_x, fish_left_x)

array([[1.        , 0.89770578],
       [0.89770578, 1.        ]])

##### Mouse Y

###### vs Right Eye

In [21]:
# Mouse X vs Right Eye
fish_mouse_y = fish[[" mouse_y"]]
fish_right_eye_y = fish[[" right_eye_y"]]

In [22]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
fish_mouse_y_values = fish_mouse_y.values
fish_mouse_y_scaled = min_max_scaler.fit_transform(fish_mouse_y_values)

# right eye x
fish_right_eye_y_values = fish_right_eye_y.values
fish_right_eye_y_scaled = min_max_scaler.fit_transform(fish_right_eye_y_values)

In [23]:
# Scipy's T-Test 
stats.ttest_ind(fish_right_eye_y_scaled, fish_mouse_y_scaled)

Ttest_indResult(statistic=array([18.5241295]), pvalue=array([8.00966252e-71]))

This t-test produces a p-value of **8.00966252e-71**, showing strong correlation between Mouse Y and Right Eye Y.

In [69]:
# Covariance
fmy_scaled = np.concatenate(fish_mouse_y_scaled, axis=0 )
frey_scaled = np.concatenate(fish_right_eye_y_scaled, axis=0 )
np.corrcoef(frey_scaled, fmy_scaled)

array([[1.        , 0.10193298],
       [0.10193298, 1.        ]])

###### vs Left Eye

In [18]:
# Mouse X vs Left Eye
fish_mouse_y = fish[[" mouse_y"]]
fish_left_eye_y = fish[[" left_eye_y"]]

In [19]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
fish_mouse_y_values = fish_mouse_y.values
fish_mouse_y_scaled = min_max_scaler.fit_transform(fish_mouse_y_values)

# right eye x
fish_left_eye_y_values = fish_left_eye_y.values
fish_left_eye_y_scaled = min_max_scaler.fit_transform(fish_left_eye_y_values)

In [20]:
# Scipy's T-Test 
stats.ttest_ind(fish_left_eye_y_scaled, fish_mouse_y_scaled)

Ttest_indResult(statistic=array([12.0495735]), pvalue=array([2.50053837e-32]))

This t-test produces a p-value of **2.50053837e-32**, showing strong correlation between Mouse Y and Left Eye Y.

In [70]:
# Covariance
fmy_scaled = np.concatenate(fish_mouse_y_scaled, axis=0 )
fley_scaled = np.concatenate(fish_left_eye_y_scaled, axis=0 )
np.corrcoef(fley_scaled, fmy_scaled)

array([[1.        , 0.08576754],
       [0.08576754, 1.        ]])

In [ ]:
# Sum of Squared Differences
ssd = np.sum((fish_right_eye_y_scaled - fish_left_eye_y_scaled)**2)
print(ssd)

9.281670508869464


In [ ]:
# Left vs Right
lr_diff = fish_right_eye_y_scaled - fish_left_eye_y_scaled
lr_diff = np.concatenate(lr_diff, axis=0 )
variance(lr_diff)

0.002207836547751615

In [24]:
# Covariance
fish_right_y = np.concatenate(fish_right_eye_y_scaled, axis=0 )
fish_left_y = np.concatenate(fish_left_eye_y_scaled, axis=0 )
np.cov(fish_right_y, fish_left_y)

array([[0.06585924, 0.0575527 ],
       [0.0575527 , 0.051454  ]])

In [25]:
# Correlation
np.corrcoef(fish_right_y, fish_left_y)

array([[1.        , 0.98866188],
       [0.98866188, 1.        ]])

#### General Use

##### Mouse X

###### vs Right Eye

In [26]:
# Mouse X vs Right Eye
gu_mouse_x = general_use[[" mouse_x"]]
gu_right_eye_x = general_use[[" right_eye_x"]]

In [27]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
gu_mouse_x_values = gu_mouse_x.values
gu_mouse_x_scaled = min_max_scaler.fit_transform(gu_mouse_x_values)

# right eye x
gu_right_eye_x_values = gu_right_eye_x.values
gu_right_eye_x_scaled = min_max_scaler.fit_transform(gu_right_eye_x_values)

In [28]:
# T-test
stats.ttest_ind(gu_right_eye_x_scaled, gu_mouse_x_scaled)

Ttest_indResult(statistic=array([5.22761196]), pvalue=array([1.89639489e-07]))

This t-test produces a p-value of **1.89639489e-07**, showing strong correlation between Mouse X and Right Eye X.

In [71]:
# Covariance
gmx_scaled = np.concatenate(gu_mouse_x_scaled, axis=0 )
grex_scaled = np.concatenate(gu_right_eye_x_scaled, axis=0 )
np.corrcoef(grex_scaled, gmx_scaled)

array([[ 1.        , -0.56426757],
       [-0.56426757,  1.        ]])

###### vs Left Eye

In [29]:
# Mouse X vs Left Eye
gu_mouse_x = general_use[[" mouse_x"]]
gu_left_eye_x = general_use[[" left_eye_x"]]

In [30]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
gu_mouse_x_values = gu_mouse_x.values
gu_mouse_x_scaled = min_max_scaler.fit_transform(gu_mouse_x_values)

# right eye x
gu_left_eye_x_values = gu_left_eye_x.values
gu_left_eye_x_scaled = min_max_scaler.fit_transform(gu_left_eye_x_values)

In [31]:
# T-test
stats.ttest_ind(gu_left_eye_x_scaled, gu_mouse_x_scaled)

Ttest_indResult(statistic=array([1.8537171]), pvalue=array([0.06392695]))

This t-test produces a p-value of **0.06392695**, showing weak correlation between Mouse X and Right Eye X.

In [72]:
# Covariance
gmx_scaled = np.concatenate(gu_mouse_x_scaled, axis=0 )
glex_scaled = np.concatenate(gu_left_eye_x_scaled, axis=0 )
np.corrcoef(glex_scaled, gmx_scaled)

array([[ 1.        , -0.56365091],
       [-0.56365091,  1.        ]])

###### Sum of Squared Differences between Left and Right

In [ ]:
# Sum of Squared Differences
ssd = np.sum((gu_right_eye_x_scaled - gu_left_eye_x_scaled)**2)
print(ssd)

1.662953887437317


In [ ]:
# Left vs Right
lr_diff = gu_right_eye_x_scaled - gu_left_eye_x_scaled
lr_diff = np.concatenate(lr_diff, axis=0 )
variance(lr_diff)

0.0007361036278139599

In [43]:
# Covariance
gu_right_x = np.concatenate(gu_right_eye_x_scaled, axis=0 )
gu_left_x = np.concatenate(gu_left_eye_x_scaled, axis=0 )
np.cov(gu_right_x, gu_left_x)

array([[0.03241693, 0.03344845],
       [0.03344845, 0.03521608]])

In [44]:
# Correlation
np.corrcoef(gu_right_x, gu_left_x)

array([[1.        , 0.98996443],
       [0.98996443, 1.        ]])

##### Mouse Y

###### vs Right Eye

In [34]:
# Mouse Y vs Right Eye
gu_mouse_y = general_use[[" mouse_y"]]
gu_right_eye_y = general_use[[" right_eye_y"]]

In [35]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse y
gu_mouse_y_values = gu_mouse_y.values
gu_mouse_y_scaled = min_max_scaler.fit_transform(gu_mouse_y_values)

# right eye y
gu_right_eye_y_values = gu_right_eye_y.values
gu_right_eye_y_scaled = min_max_scaler.fit_transform(gu_right_eye_y_values)

In [36]:
# T-test
stats.ttest_ind(gu_right_eye_y_scaled, gu_mouse_y_scaled)

Ttest_indResult(statistic=array([-0.77155505]), pvalue=array([0.44046925]))

This t-test produces a p-value of **0.44046925**, showing weak correlation between Mouse Y and Right Eye Y.

In [73]:
# Covariance
gmy_scaled = np.concatenate(gu_mouse_y_scaled, axis=0 )
grey_scaled = np.concatenate(gu_right_eye_y_scaled, axis=0 )
np.corrcoef(grey_scaled, gmy_scaled)

array([[1.        , 0.47535246],
       [0.47535246, 1.        ]])

###### vs Left Eye

In [37]:
# Mouse Y vs Left Eye
gu_mouse_y = general_use[[" mouse_y"]]
gu_left_eye_y = general_use[[" left_eye_y"]]

In [38]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse y
gu_mouse_y_values = gu_mouse_y.values
gu_mouse_y_scaled = min_max_scaler.fit_transform(gu_mouse_y_values)

# right eye y
gu_left_eye_y_values = gu_left_eye_y.values
gu_left_eye_y_scaled = min_max_scaler.fit_transform(gu_left_eye_y_values)

In [39]:
# T-test
stats.ttest_ind(gu_left_eye_y_scaled, gu_mouse_y_scaled)

Ttest_indResult(statistic=array([0.79924006]), pvalue=array([0.42424623]))

This t-test produces a p-value of **0.42424623**, showing weak correlation between Mouse Y and Left Eye Y.

In [74]:
# Covariance
gmy_scaled = np.concatenate(gu_mouse_y_scaled, axis=0 )
gley_scaled = np.concatenate(gu_left_eye_y_scaled, axis=0 )
np.corrcoef(gley_scaled, gmy_scaled)

array([[1.        , 0.63975579],
       [0.63975579, 1.        ]])

In [ ]:
# Sum of Squared Differences
ssd = np.sum((gu_right_eye_y_scaled - gu_left_eye_y_scaled)**2)
print(ssd)

7.087775796010029


In [ ]:
# Left vs Right
lr_diff = gu_right_eye_y_scaled - gu_left_eye_y_scaled
lr_diff = np.concatenate(lr_diff, axis=0 )
variance(lr_diff)

0.006717151158292652

In [42]:
# Covariance
gu_right_y = np.concatenate(gu_right_eye_y_scaled, axis=0 )
gu_left_y = np.concatenate(gu_left_eye_y_scaled, axis=0 )
np.cov(gu_right_y, gu_left_y)

array([[0.02270403, 0.01812224],
       [0.01812224, 0.02025759]])

In [45]:
# Correlation
np.corrcoef(gu_right_y, gu_left_y)

array([[1.        , 0.84501884],
       [0.84501884, 1.        ]])

#### OSU

##### Mouse X

###### vs Right Eye

In [46]:
# Mouse X vs Right Eye
osu_mouse_x = general_use[[" mouse_x"]]
osu_right_eye_x = general_use[[" right_eye_x"]]

In [47]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
osu_mouse_x_values = osu_mouse_x.values
osu_mouse_x_scaled = min_max_scaler.fit_transform(osu_mouse_x_values)

# right eye x
osu_right_eye_x_values = osu_right_eye_x.values
osu_right_eye_x_scaled = min_max_scaler.fit_transform(osu_right_eye_x_values)

In [48]:
# Sum of Squared Differences
ssd = np.sum((osu_right_eye_x_scaled - osu_mouse_x_scaled)**2)
print(ssd)

130.84665880320222


In [49]:
# T-test
stats.ttest_ind(osu_right_eye_x_scaled, osu_mouse_x_scaled)

Ttest_indResult(statistic=array([5.22761196]), pvalue=array([1.89639489e-07]))

This t-test produces a p-value of **1.89639489e-07**, showing strong correlation between Mouse X and Right Eye X.

In [75]:
# Covariance
omx_scaled = np.concatenate(osu_mouse_x_scaled, axis=0 )
orex_scaled = np.concatenate(osu_right_eye_x_scaled, axis=0 )
np.corrcoef(orex_scaled, omx_scaled)

array([[ 1.        , -0.56426757],
       [-0.56426757,  1.        ]])

###### vs Left Eye

In [50]:
# Mouse X vs Left Eye
osu_mouse_x = general_use[[" mouse_x"]]
osu_left_eye_x = general_use[[" left_eye_x"]]

In [51]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
osu_mouse_x_values = osu_mouse_x.values
osu_mouse_x_scaled = min_max_scaler.fit_transform(osu_mouse_x_values)

# right eye x
osu_left_eye_x_values = osu_left_eye_x.values
osu_left_eye_x_scaled = min_max_scaler.fit_transform(osu_left_eye_x_values)

In [52]:
# Sum of Squared Differences
ssd = np.sum((osu_left_eye_x_scaled - osu_mouse_x_scaled)**2)
print(ssd)

133.5478112571949


In [53]:
# T-test
stats.ttest_ind(osu_left_eye_x_scaled, osu_mouse_x_scaled)

Ttest_indResult(statistic=array([1.8537171]), pvalue=array([0.06392695]))

This t-test produces a p-value of **0.06392695**, showing strong correlation between Mouse X and Left Eye X.

In [76]:
# Covariance
omx_scaled = np.concatenate(osu_mouse_x_scaled, axis=0 )
olex_scaled = np.concatenate(osu_left_eye_x_scaled, axis=0 )
np.corrcoef(olex_scaled, omx_scaled)

array([[ 1.        , -0.56365091],
       [-0.56365091,  1.        ]])

In [ ]:
# Sum of Squared Differences
ssd = np.sum((osu_right_eye_x_scaled - osu_left_eye_x_scaled)**2)
print(ssd)

1.662953887437317


In [ ]:
# Left vs Right
lr_diff = osu_right_eye_x_scaled - osu_left_eye_x_scaled
lr_diff = np.concatenate(lr_diff, axis=0 )
variance(lr_diff)

0.0007361036278139599

In [54]:
# Covariance
osu_right_x = np.concatenate(osu_right_eye_x_scaled, axis=0 )
osu_left_x = np.concatenate(osu_left_eye_x_scaled, axis=0 )
np.cov(osu_right_x, osu_left_x)

array([[0.03241693, 0.03344845],
       [0.03344845, 0.03521608]])

In [55]:
# Correlation
np.corrcoef(osu_right_x, osu_left_x)

array([[1.        , 0.98996443],
       [0.98996443, 1.        ]])

##### Mouse Y

###### vs Right Eye

In [56]:
# Mouse Y vs Right Eye Y
osu_mouse_y = general_use[[" mouse_y"]]
osu_right_eye_y = general_use[[" right_eye_y"]]

In [57]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
osu_mouse_y_values = osu_mouse_y.values
osu_mouse_y_scaled = min_max_scaler.fit_transform(osu_mouse_y_values)

# right eye x
osu_right_eye_y_values = osu_right_eye_y.values
osu_right_eye_y_scaled = min_max_scaler.fit_transform(osu_right_eye_y_values)

In [58]:
# Sum of Squared Differences
ssd = np.sum((osu_right_eye_y_scaled - osu_mouse_y_scaled)**2)
print(ssd)

102.62080759291057


In [59]:
# T-test
stats.ttest_ind(osu_right_eye_y_scaled, osu_mouse_y_scaled)

Ttest_indResult(statistic=array([-0.77155505]), pvalue=array([0.44046925]))

This t-test produces a p-value of **0.44046925**, showing strong correlation between Mouse X and Left Eye X.

In [77]:
# Covariance
omy_scaled = np.concatenate(osu_mouse_y_scaled, axis=0 )
orey_scaled = np.concatenate(osu_right_eye_y_scaled, axis=0 )
np.corrcoef(orey_scaled, omy_scaled)

array([[1.        , 0.47535246],
       [0.47535246, 1.        ]])

###### vs Left Eye

In [60]:
# Mouse Y vs Left Eye Y
osu_mouse_y = general_use[[" mouse_y"]]
osu_left_eye_y = general_use[[" left_eye_y"]]

In [61]:
# Normalize feature columns
min_max_scaler = preprocessing.MinMaxScaler()

# mouse x
osu_mouse_y_values = osu_mouse_y.values
osu_mouse_y_scaled = min_max_scaler.fit_transform(osu_mouse_y_values)

# right eye x
osu_left_eye_y_values = osu_left_eye_y.values
osu_left_eye_y_scaled = min_max_scaler.fit_transform(osu_left_eye_y_values)

In [62]:
# Sum of Squared Differences
ssd = np.sum((osu_left_eye_y_scaled - osu_mouse_y_scaled)**2)
print(ssd)

84.25461987793591


In [63]:
# T-test
stats.ttest_ind(osu_left_eye_y_scaled, osu_mouse_y_scaled)

Ttest_indResult(statistic=array([0.79924006]), pvalue=array([0.42424623]))

This t-test produces a p-value of **0.42424623**, showing strong correlation between Mouse X and Left Eye X.

In [78]:
# Covariance
omy_scaled = np.concatenate(osu_mouse_y_scaled, axis=0 )
oley_scaled = np.concatenate(osu_left_eye_y_scaled, axis=0 )
np.corrcoef(oley_scaled, omy_scaled)

array([[1.        , 0.63975579],
       [0.63975579, 1.        ]])

In [ ]:
# Sum of Squared Differences
ssd = np.sum((osu_right_eye_y_scaled - osu_left_eye_y_scaled)**2)
print(ssd)

7.087775796010029


In [ ]:
# Left vs Right
lr_diff = osu_right_eye_y_scaled - osu_left_eye_y_scaled
lr_diff = np.concatenate(lr_diff, axis=0 )
variance(lr_diff)

0.006717151158292652

In [64]:
# Covariance
osu_right_y = np.concatenate(osu_right_eye_y_scaled, axis=0 )
osu_left_y = np.concatenate(osu_left_eye_y_scaled, axis=0 )
np.cov(osu_right_y, osu_left_y)

array([[0.02025759, 0.01812224],
       [0.01812224, 0.02270403]])

In [65]:
# Correlation
np.corrcoef(osu_right_y, osu_left_y)

array([[1.        , 0.84501884],
       [0.84501884, 1.        ]])